client/src/main/scala/io/mediachain/client/MediachainCopycatClient.scala

```diff
@@ -0,0 +1,155 @@
+package io.mediachain.client
+
+import cats.data.XorT
+import io.mediachain.protocol.Datastore.{JournalEntry, _}
+import io.mediachain.protocol.Transactor.JournalListener
+import io.mediachain.copycat.Client.{ClientState, ClientStateListener}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+
+class MediachainCopycatClient(datastore: Datastore)
+  (implicit executionContext: ExecutionContext = ExecutionContext.global)
+  extends MediachainClient with JournalListener with ClientStateListener
+{
+
+  def allCanonicalReferences = canonicalRefs
+
+  def chainForCanonical(ref: Reference): Future[Option[Reference]] = {
+    // TODO: handle disconnected cluster state
```

Comment:
yeah, the only reasonable reaction to disconnecting is to complain loudly and back-off retry. And if you keep failing for too long give up altogether.